lib/preview-web/src/Preview.tsx

```diff
@@ -252,7 +252,7 @@ export class Preview<TFramework extends AnyFramework> {
     const render = this.storyRenders.find((r) => r.id === storyId);
     const story = render?.story || (await this.storyStore.loadStory({ storyId }));
 
-    const argNamesToReset = argNames || Object.keys(this.storyStore.args.get(storyId));
+    const argNamesToReset = argNames || Object.keys(story.initialArgs);
```

Comment:
@tmeasday could you take a look at this change? Thanks!

Reply:
Hmm, this doesn't seem right.

This change ensures we always reset any arg that is initially set, but doesn't ensure we reset args that are set that weren't set initially.

I am thinking the reason this works is because if `foo=!undefined` we end up *unsetting* `foo`, in which case:
 1. `foo` (may have been) initially set, so is in `Object.keys(story.initialArgs)`
 2. `foo` is no longer set, so is not in `Object.keys(this.storyStore.args.get(storyId)`

Thus changing the code to check 1 but not 2 will reset `foo`.

However, it introduces a new bug, as evidenced by the test case below, that if an arg is set but *wasn't* set in `initialArgs`, it will never get reset.

Maybe we want something like `[...new Set([...Object.keys(story.initialArgs), ...Object.keys(this.storyStore.args.get(storyId))])]`?

Reply:
Thank you for clarifying this! I will revisit this shortly. Thanks!